docker-compose.yml

```diff
@@ -2,38 +2,37 @@ version: '3.0'
 
 services:
   web:
-    build: .
-    entrypoint: >
-      bash -c "
-      rm -f tmp/pids/server.pid
-      && bundle exec rails s -b 0.0.0.0 -p 3000"
+    build:
+      context: .
+      dockerfile: Dockerfile.dev
     ports:
-      - ${RAILS_PORT-3000}:3000
+      - 3000:3000
     environment:
-      POSTGRES_USER: ${POSTGRES_USER-postgres}
-      POSTGRES_PASSWORD: ${POSTGRES_PASSWORD-postgres}
+      POSTGRES_USER: postgres
       POSTGRES_HOST: db
-    volumes:
-      - .:/src/myapp
+    tty: true
+    stdin_open: true
+    develop:
+      watch:
+        - action: sync
+          path: .
+          target: /src/app
+        - action: rebuild
+          path: package.json
+        - action: rebuild
+          path: Gemfile
     depends_on:
       - db
     links:
       - db:db
   db:
     image: postgres:15
     environment:
-      POSTGRES_USER: ${POSTGRES_USER-postgres}
-      POSTGRES_PASSWORD: ${POSTGRES_PASSWORD-postgres}
+      POSTGRES_USER: postgres
+      POSTGRES_HOST_AUTH_METHOD: trust
     volumes:
       - db_data:/var/lib/postgresql/data
     ports:
-      - ${POSTGRES_PORT-5432}:5432
-  js:
-    build: .
-    entrypoint: yarn build --watch
-    volumes:
-      - .:/src/myapp
-      - node_modules:/src/myapp/node_modules
+      - 5432:5432
 volumes:
   db_data:
-  node_modules:
```

Comment:
just curious but why this removal? wasn't it trying to cache the downloaded node_modules?

Reply:
Two things to note here:

1 - We are using the [docker-compose watch](https://docs.docker.com/compose/file-watch/) feature that re-builds the image if the `package.json` file is updated
```YAML
develop:
  watch:
    - action: rebuild
      path: package.json
```

2 - The `dockerfile.dev` file has `CMD ["./bin/dev"]`. This means that by default (and we are not overriding this in the compose file) it will run that script when you start a container from the built image. This scripts uses foreman and run both the Rails server and a node watch script.


So the flow would be like:
1. First time you build and run the compose services: Image is created using the `dockerfile.dev` file and both Rails and Node watch process are executed
2. If you update your dependencies on the `package.json` file, step 1 is executed again, so new/updated libraries are refreshed and the watch process updates the JS code.